libraries/chain/block_log.cpp

```diff
@@ -286,28 +296,76 @@ namespace eosio { namespace chain {
       }
    }
 
+   std::vector<char>  detail::block_log_impl::pack(const pruned_block& b, pruned_transaction::cf_compression_type segment_compression) {
+
+      // In version 4 of the irreversible blocks log format, these log entries consists of the following in order:
+      //    1. An uint32_t offset from the start of this log entry to the start of the next log entry.
+      //    2. An uint32_t indicating the compression status for the serialization of the pruned_block following this.
+      //    3. The serialization of a pruned_block representation of the block for the entry including padding.
+
+      std::size_t padded_size = b.maximum_pruned_pack_size(segment_compression);
+      std::vector<char> buffer(padded_size + 2* sizeof(uint32_t));
```

Comment:
This would be adding the `sizeof(uint32_t)` once and then also the `pack_size` of the `segment_compression` value as a `fc::unsigned_int` (currently should always be 1 but don't hard code).